src/Init/Data/BitVec/Bitblast.lean

```diff
@@ -326,4 +326,101 @@ theorem getLsb_mul (x y : BitVec w) (i : Nat) :
   · simp
   · omega
 
+/-## shiftLeft recurrence for bitblasting -/
+
+/--
+A recurrence that describes shifting by an arbitrary bitvector
+as shifting by a constant amount.
+ -/
+def shiftLeftRec (x : BitVec w₁) (y : BitVec w₂) (n : Nat) : BitVec w₁ :=
+  let shiftAmt := (y &&& (twoPow w₂ n))
+  match n with
+  | 0 => x <<< shiftAmt
+  | n + 1 => (shiftLeftRec x y n) <<< shiftAmt
```

Comment:
From a quick glance, it seems `shiftLeftRec x y n` computes "shifting `x` left by the first `n` bits of `y`", i.e:
  `shiftLeftRec x y n = x <<< (y.toNat % 2^n)`
I think it'd be helpful to at least write this direct form in the docstring